terraform/modules/networking/vpc_endpoints.tf

```diff
@@ -48,6 +47,21 @@ resource "aws_vpc_endpoint" "vpc_endpoints" {
   subnet_ids          = aws_subnet.private_subnet[*].id
   security_group_ids  = [aws_security_group.aws_service_vpc_endpoints.id]
   private_dns_enabled = true
+}
+
+# s3 requires additional options due to there also being a 'gateway' VPC endpoint for s3
+data "aws_vpc_endpoint_service" "s3" {
+  service      = "s3"
+  service_type = "Interface"
+}
+
+resource "aws_vpc_endpoint" "s3_endpoint" {
+  vpc_id              = aws_vpc.main.id
+  service_name        = data.aws_vpc_endpoint_service.s3.service_name
+  vpc_endpoint_type   = "Interface"
+  subnet_ids          = aws_subnet.private_subnet[*].id
+  security_group_ids  = [aws_security_group.aws_service_vpc_endpoints.id]
+  private_dns_enabled = true
   dns_options {
     private_dns_only_for_inbound_resolver_endpoint = false
   }
```

Comment:
So is S3 the only endpoint that has the buggy default value for `private_dns_only_for_inbound_resolver_endpoint`?